modules/siwe.ts

```diff
@@ -48,68 +50,119 @@ class Siwe {
       _id: walletAddress
     };
     const newEthAccountXhr = await privateBaseAxios.post('/account/ethereum/create', data);
-    console.log({ newEthAccountXhr })
     const newEthAccount = newEthAccountXhr.data
 
     /* Update user_id of newEthAccount */
     const updateData = {
       user_id: userId
     }
-    console.log({ updateData })
     const updatedAccountXhr = await privateBaseAxios.patch(`/account/ethereum/update/${newEthAccount._id}`, updateData)
     return updatedAccountXhr.data
   }
 
-  static async signInWithEthereum(discordId: string) {
+  static encodeAddress (address: string): string {
+    const addr = address.toLowerCase().replace('0x', '');
+    const hash = keccak('keccak256').update(addr).digest('hex');
+    let ret = '0x';
 
-    console.log({ signer, provider })
-
-    if (!signer) { alert('Wallet not connected!'); return }
-
-    const walletAddress = await signer.getAddress();
-
-    console.log({ walletAddress })
-
-    /* Get nonce from server */
-    const nonceRes = await privateBaseAxios.get(`/siwe/nonce/${walletAddress}`)
-
-    const nonce = nonceRes.data
-
-    console.log({ nonce })
-
-    const message = this.createSiweMessage(
-      walletAddress,
-      `Sign in with Ethereum to the app. - link to discord_id ${discordId}`,
-      nonce
-    );
+    for (let i = 0; i < addr.length; i++) {
```

Comment:
can we maybe use ethers for this? Not that I'm doubting it works :) `ethers.utils.getAddress(address)` should do the same

Reply:
I'll look at that. I plan on cleaning up `siwe.ts` in general.

Reply:
yea this seems like something a util would exist for

Reply:
I switched to the util